src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -424,10 +439,11 @@ private void DoStart()
 
             // Set up an execution context for this thread so that the target creation steps can use asynch values.
             RuntimeContext.InitializeMainThread();
-            
+
             // Initialize the implicit stream subscribers table.
             var implicitStreamSubscriberTable = Services.GetRequiredService<ImplicitStreamSubscriberTable>();
-            implicitStreamSubscriberTable.InitImplicitStreamSubscribers(this.grainTypeManager.GrainClassTypeData.Select(t => t.Value.Type).ToArray());
+            var grainTypeManager = Services.GetRequiredService<GrainTypeManager>();
```

Comment:
Sorry to keep pushing on this, but it doesn't feel completely safe (at least by just browsing the code here) to use the grain type manager when it might still be starting in parallel. In the past we were sure that by the time we try to use it, it was already started. Here we are relying on luck a little bit, aren't we?

Reply:
I don't see how?  This logic runs in RuntimeServices stage, which comes after RuntimeInitialize sage, which is where the grainTypeManager is started.  Each stage is performed in parallel, but the stages are run in order.  I don't see how we could get here without the grainTypeManager being started.  Am I missing something?